### ISSUE TITLE
Error 1011 / hipErrorInvalidValue during hipMemset

### ISSUE THREAD

**Issue by davidrohr** [ROCm-Developer-Tools/HIP#1131]
I have compiled ROCm and HIP from sources for a Gentoo Linux 64 bit operating system. HIP seems to work except for hipMemset. The attached example code runs fine on a ubuntu machine, but on the Gentoo system the hipMemset fails. Only hipMemset fails, hipMemcpy and kernel execution run perfectly. Attached is the testcase I am using.
[sample.tar.gz](https://github.com/ROCm-Developer-Tools/HIP/files/3211083/sample.tar.gz)

I know gentoo is not officially supported, but I am wondering why only hipMemset fails while the rest work, and I'd appreciate any idea.
Compile command is: hipcc --amdgpu-target=gfx906 -o test2 test2.hip.cpp

**Comment by bensander** [comment]
Hipmemset uses an internal kernel written in HCC language iirc.  Probably could be changed to use the rocr memset api.  Are you linking with HCC?

**Comment by gargrahul** [comment]
We may need more debug detail. Since you are able to launch kernels in your shared test, could the issue be usage of __builtin_memcpy in hipmemset kernel?
https://github.com/ROCm-Developer-Tools/HIP/blob/master/src/hip_memory.cpp#L1481

**Comment by davidrohr** [comment]
I do not think using a kernel is a bad idea in general, in particular as it works well on one of my systems. I am just wondering why it fails on the other.

So what debug info do you need? I could create a kernel myself that uses __builtin_memcpy, and try if that works. Would that help?

**Comment by gargrahul** [comment]
Yes. Trying with your custom kernel with __builtin_memcpy should help.
Most likely it is throwing an exception in https://github.com/ROCm-Developer-Tools/HIP/blob/master/src/hip_memory.cpp#L1539 , if you can check exception value, that may be useful too.

**Comment by davidrohr** [comment]
OK, thx. I am out of office today, will report back on Monday.

**Comment by davidrohr** [comment]
I spent some more time debugging. The problem is not with __builtin_memcpy.
In the attached testcase, I have copy and pasted in the ihipMemset from HIP, and that seems to work.
[sample.tar.gz](https://github.com/ROCm-Developer-Tools/HIP/files/3223351/sample.tar.gz)

The "hipError_t e = ihipMemset((char*) devPtr, 0, SIZE, stream, ihipMemsetDataTypeChar);" in my sample works correctly, cleaning the memory, but the hipMemset(devPtr, 0, SIZE) fails.

I have rebuilt HIP with debug symbols and ran it through gdb.
https://github.com/ROCm-Developer-Tools/HIP/blob/master/src/hip_memory.cpp#L1552 throws an exception, but ex.what() yields "std::exception" only. 

Looking in more detail, the exception comes from https://github.com/ROCm-Developer-Tools/HIP/blob/master/include/hip/hcc_detail/functional_grid_launch.hpp, namely the function call tohip_impl::make_kernarg in hipLaunchKernelGGL.

I do not understand why this fails in hipMemset, but not in the ihipMemset which I copy and pasted into my sample.
